### PR TITLE
Allow tcp forwarding

### LINKED ISSUE #5527: 5.2.17 Ensure SSH AllowTcpForwarding is disabled (Scored)
Needs rule

### PR DESCRIPTION
**Pull request by eradot4027**
#### Description:

- The AllowTcpForwarding parameter specifies the maximum number of open sessions permitted from a given connection. To disable AllowTcpForwarding, add or correct the following line in /etc/ssh/sshd_config: AllowTcpForwarding no

#### Rationale:

- Leaving port forwarding enabled can expose the organization to security risks and back-doors.

- Fixes #5527

### CODE REVIEW

File: linux_os/guide/services/ssh/ssh_server/sshd_disable_allow_tcp_forwarding/tests/comment.pass.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+#
+# profiles = xccdf_org.ssgproject.content_profile_ospp
```
Comment (vojtapolasek) [inline linux_os/guide/services/ssh/ssh_server/sshd_disable_allow_tcp_forwarding/tests/comment.pass.sh:3]: please remove these #profiles lines from all tests. They should not be there if the test is not specific for certain profile. In this case I think it is not.

File: linux_os/guide/services/ssh/ssh_server/sshd_disable_allow_tcp_forwarding/rule.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    following line in <tt>/etc/ssh/sshd_config</tt>:
+    <pre>AllowTcpForwarding no</pre>
+
+rationale: |-
+    Leaving port forwarding enabled can expose the organization to security risks and back-doors.
+
+severity: medium
+
+references:
+  cis@rhel8: 5.2.16
```
[the remark this answers is not in the document]
Comment (eradot4027) [reply linux_os/guide/services/ssh/ssh_server/sshd_disable_allow_tcp_forwarding/rule.yml:17]: Sorry, I was looking at the new draft.

File: linux_os/guide/services/ssh/ssh_server/sshd_disable_allow_tcp_forwarding/rule.yml [hunk 3]
```diff
@@ -3,8 +3,8 @@ documentation_complete: true
 title: 'Disable AllowTcpForwarding'
 
 description: |-
-    The <tt>AllowTcpForwarding</tt> parameter specifies the maximum number of open sessions permitted
-    from a given connection. To disable AllowTcpForwarding, add or correct the
+    The <tt>AllowTcpForwarding</tt> parameter specifies whether TCP forwarding is permitted.
+    To disable AllowTcpForwarding, add or correct the
```
Comment (vojtapolasek) [inline linux_os/guide/services/ssh/ssh_server/sshd_disable_allow_tcp_forwarding/rule.yml:7]:
```suggestion
    To disable TCP forwarding, add or correct the
```

**Comment by vojtapolasek** [comment]
If you don't mind, I would also rename the whole rule only to sshd_disable_tcp_forwarding. disable_allow_tcp_forwarding sounds a bit strange. I checked other sshd rules and many of them don't keep exact option name in their name. I mean renaming the folder.

File: linux_os/guide/services/ssh/ssh_server/sshd_disable_allow_tcp_forwarding/rule.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+documentation_complete: true
+
+title: 'Disable AllowTcpForwarding'
```
Comment (vojtapolasek) [inline linux_os/guide/services/ssh/ssh_server/sshd_disable_allow_tcp_forwarding/rule.yml:3]:
```suggestion
title: 'Disable SSH TCP forwarding '
```

**Comment by vojtapolasek** [comment]
Thank you, good job.

**Pull request merged**